File: js/src/ArrayProblem/GasStation.js


export const canCompleteCircuit = (gas, cost) =>{


		let totalCost = 0;

		let totalGas = 0;

		let currentGas = 0;

		let currentStation = 0;

		const length = gas.length;


		for (let i = 0; i < length; i++) {
			
			totalCost += cost[i];

			totalGas += gas[i];
		}

		if (totalGas < totalCost) return -1;


		for (let i =0; i< length; i++) {
			
			currentGas += gas[i] - cost[i];

			if (currentGas < 0) {

				currentGas = 0;

				currentStation = i + 1;
			}
		}

		return currentStation;
}

File: python/array_problem/gas_station.py



def can_complete_circuit(gas, cost):

	total_gas = 0

	total_cost = 0

	current_station = 0

	current_gas = 0

	length = len(gas)


	for k in range(length):

		total_gas += gas[k]

		total_cost += cost[k]


	if total_gas < total_cost:
		
		return -1

	for i in range(length):
		
		current_gas += gas[i] - cost[i]

		if current_gas < 0:
			
			current_gas = 0

			current_station = i + 1


	return current_station